From: Skydio <noreply@skydio.com>
Date: Mon, 20 Jul 2026 00:00:00 +0000
Subject: [PATCH] logger: Use fstring::get() instead of the deprecated conversion

fmt 12 marked fmt::fstring's implicit conversion to string_view as deprecated in
favor of fstring::get(). logger::log() relies on that conversion to pass a
format_string into log_(), which takes a string_view, so every caller trips
-Werror=deprecated-declarations.

Call get() explicitly at the two sites that convert. All other format_string
overloads forward the type along and are unaffected.
---
 include/spdlog/logger.h | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/include/spdlog/logger.h b/include/spdlog/logger.h
--- a/include/spdlog/logger.h
+++ b/include/spdlog/logger.h
@@ -80,7 +80,7 @@
     template<typename... Args>
     void log(source_loc loc, level::level_enum lvl, fmt::format_string<Args...> fmt, Args &&...args)
     {
-        log_(loc, lvl, fmt, std::forward<Args>(args)...);
+        log_(loc, lvl, fmt.get(), std::forward<Args>(args)...);
     }

     template<typename... Args>
@@ -186,7 +186,7 @@
     template<typename... Args>
     void log(source_loc loc, level::level_enum lvl, fmt::wformat_string<Args...> fmt, Args &&...args)
     {
-        log_(loc, lvl, fmt, std::forward<Args>(args)...);
+        log_(loc, lvl, fmt.get(), std::forward<Args>(args)...);
     }

     template<typename... Args>
